sqlalchemy_ibmi/backend.py

```diff
@@ -105,25 +155,31 @@ def create_connect_args(self, url):
         opts.update(url.query)
 
         keys = opts
-        query = url.query
 
         connect_args = {}
         for param in ('ansi', 'unicode_results', 'autocommit'):
             if param in keys:
                 connect_args[param] = util.asbool(keys.pop(param))
 
         if 'odbc_connect' in keys:
-            connectors = [urllib.unquote_plus(keys.pop('odbc_connect'))]
+            connectors = [urllib.parse.unquote_plus(keys.pop('odbc_connect'))]
         else:
             dsn_connection = 'dsn' in keys or \
                              ('host' in keys and 'database' not in keys)
             if dsn_connection:
-                connectors = ['dsn=%s' % (keys.pop('host', '') or \
+                connectors = ['dsn=%s' % (keys.pop('host', '') or
                                           keys.pop('dsn', ''))]
             else:
-                connectors = ["DRIVER={%s}" % keys.pop('driver', self.pyodbc_driver_name),
-                              'System=%s' % keys.pop('host', ''),
-                              'DBQ=QGPL']
+                connectors = [
+                    "DRIVER={%s}" %
+                    keys.pop(
+                        'driver',
+                        self.pyodbc_driver_name),
+                    'System=%s' %
+                    keys.pop(
+                        'host',
+                        ''),
+                    'DBQ=QGPL']
```

Comment:
The auto formatting on this did quite the horrific job, but we need to refactor this code anyway, so I'm ok with it.